proof_of_concept/rest/client.py

```diff
@@ -52,6 +53,35 @@ def retrieve_asset(self, site_id: Identifier, asset_id: Identifier
 
         raise ValueError(f'Site {site_id} does not have a store')
 
+    def retrieve_asset_image(self, asset_location: str, target: Path) -> None:
+        """Obtains an asset image from a store.
+
+        This downloads the image at the given location into a file at
+        the given path.
+
+        Args:
+            asset_location: URL of the image to download.
+            target: Path of the file to save.
+        """
+        with requests.get(
+                asset_location,
+                params={'requester': self._site},
+                stream=True) as r:
+            if r.status_code == 404:
+                raise KeyError('Asset image not found')
+            elif not r.ok:
+                raise RuntimeError('Server error when retrieving asset image')
+
+            if r.headers.get('Transfer-Encoding', '') == 'chunked':
+                chunk_size = None
+            else:
+                chunk_size = 1024 * 1024
```

Comment:
You could extract a constant

Reply:
Hm.
```
if r.headers...:
    chunk_size = None
else:
    chunk_size = CHUNK_SIZE
```
I vaguely recall playing with chunk sizes when I was optimising transfer speed in cerulean, and I don't think the exact size mattered very much there, increasing the number of simultaneous requests in flight was what did the trick. So it's a technical detail that I don't expect to ever need to fiddle with again.

Reply:
Bueno